test/OrchardCore.Commerce.Tests.UI/Tests/PromotionTests/PromotionBehaviourTests.cs

```diff
@@ -35,4 +36,33 @@ async Task CheckDiscountPresenceAsync(bool hasNewPrice)
                 await CheckDiscountPresenceAsync(hasNewPrice: true);
             },
             browser);
+
+    [Theory, Chrome]
+    public Task DiscountShouldProperlyAppearOnDiscountedProduct(Browser browser) =>
+        ExecuteTestAfterSetupAsync(
+            async context =>
+            {
+                // Discount should appear on discounted product's page.
+                await context.GoToRelativeUrlAsync("/testdiscountedproduct");
+                context.Get(By.CssSelector(".price-part-price-field-value > del")).Text.ShouldBe("$5.00");
+                context.Get(By.CssSelector(".price-part-price-field-value > span")).Text.ShouldBe("$3.00");
+                await context.ClickReliablyOnSubmitAsync();
+
```

Comment:
Please also check for the Old Price on the cart page, since it seems like it's still not displayed: 
![oldprice](https://github.com/OrchardCMS/OrchardCore.Commerce/assets/80963259/c0458b10-57b3-4d08-a7dd-0fd395d6df2e)

Reply:
That's a separate problem, but I'll take a look if there's a quick fix.

Reply:
Ah okay. I didn't see an issue about this though. So if you want you can fix it here, or create an issue.

Reply:
Addressed it here.